core/src/main/scala/org/apache/spark/status/AppStatusListener.scala

```diff
@@ -211,6 +211,11 @@ private[spark] class AppStatusListener(
     updateBlackListStatus(event.executorId, true)
   }
 
+  override def onExecutorBlacklistedForStage(
+    event: SparkListenerExecutorBlacklistedForStage): Unit = {
+    updateBlackListStatusForStage(event.executorId, event.stageId, event.stageAttemptId)
+  }
+
```

Comment:
Consider covering this functionality (updating the status) in AppStatusListenerSuite. We already have a check for blacklisting an executor, we should have the same for a stage.

Reply:
Good catch. I will add the test tomorrow.